Src/Particle/AMReX_Particle.H

```diff
@@ -67,51 +67,61 @@ struct ParticleIDWrapper
     AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
     ParticleIDWrapper& operator= (const Long id) noexcept
     {
-        // zero out the 40 leftmost bits, which store the sign and the abs of the id;
+        // zero out the 40 leftmost bits, store the id including its sign bit (0: +, 1: -)
         m_idata &= 0x00FFFFFF;
 
-        uint64_t val;
-        uint64_t sign = id >= 0;
-        if (sign)
-        {
-            // 2**39-1, the max value representable in this fashion
-            AMREX_ASSERT(id <= 549755813887L);
-            val = id;
-        }
-        else
-        {
-            // -2**39-1, the min value representable in this fashion
-            AMREX_ASSERT(id >= -549755813887L);
-            val = -id;
-        }
-
-        m_idata |= (sign << 63);  // put the sign in the leftmost bit
-        m_idata |= (val << 24);   // put the val in the next 39
+        // +/-2**39-1, the max value representable in this fashion
+        AMREX_ASSERT(std::abs(id) <= 549755813887L);
+
+        m_idata |= (id << 24);   // put the id in the 40 uppermost bits
```

Comment:
According to https://en.cppreference.com/w/cpp/language/operator_arithmetic#Bitwise_shift_operators
for negative id, `id << 24` is undefined behavior.

Reply:
Ooops, I forgot to xor in and out the sign bit before C++20 (as done in long below)

Reply:
Ah, and even with that, this only works for positive id assignments.